### DISCUSSION TITLE
error on converting Inverntory module from odoo 15 to odoo 16

### DISCUSSION THREAD

**Discussion by arul-ganapathy-appasamy** [OCA/OpenUpgrade#3910] [Q&A]
psycopg2.errors.ForeignKeyViolation: insert or update on table "stock_route_warehouse" violates foreign key constraint "stock_route_warehouse_route_id_fkey"
DETAIL:  Key (route_id)=(2) is not present in table "stock_route".

This is the error that i got on converting my empty data module from( odoo 15 to odoo 16) that contain inventory module, when i try to convert my module i got this error, can any one give solution for this error.

And also i want to convert my module with data, please help me.

![Screenshot from 2023-06-07 16-56-09](https://github.com/OCA/OpenUpgrade/assets/65151635/18934123-b0fd-4a8f-8c69-331ec9d5fa78)

**Comment by legalsylvain** [comment]
Hi @arul-ganapathy-appasamy 
You try to migrate a database with ``stock`` module installed. However, OpenUpgrade doesn't handle ``stock`` migration for the time being.
Please : 
- Refer to the documentation to compare the modules installed on your instance, with the module coverage of OpenUpgrade : https://oca.github.io/OpenUpgrade/status.html 
- review PR https://github.com/OCA/OpenUpgrade/issues/3681.
- read the documentation https://oca.github.io/OpenUpgrade and contribute making PRs.

Kind regards.

(closing, as it is an expected error)

**Reply by arul-ganapathy-appasamy** [reply]
Thanks for your reply.. Avatar Sylvain LE GAL
